Goods

k1llama opened the pull request.

k1llama committed:
Create project, add models, repositories, services and controllers

anton-beetroot-23071996 commented on App.Goods/Repositories/ProductsRepository.cs:
```diff
@@ -0,0 +1,51 @@
+            _products = ProductInitializer.Initialize();
+        }
+
+        public void Add(Product item)
```
according to the task, adding of products is not supported

anton-beetroot-23071996 commented on App.Goods/Services/OrdersManager.cs:
```diff
@@ -0,0 +1,40 @@
+
+        public IEnumerable<Order> GetAllOrders() => _ordersRepository.GetAll();
+
+        public void MakeOrder(int[] products)
```
Maybe this operation should return identifier of created order ?

anton-beetroot-23071996 commented on App.Goods/Services/ProductsManager.cs:
```diff
@@ -0,0 +1,17 @@
+
+        public ProductsManager(IRepository<Product> productsRepository) => _productsRepository = productsRepository;
+
+        public IEnumerable<Product> GetAllGoods() => _productsRepository.GetAll();
```
GetAllGoods => GetAllProducts. As you use term "Products" everywhere, you should remove term "Goods" to supply consistency

anton-beetroot-23071996 commented on App.Goods/Services/ValidateService.cs:
```diff
@@ -0,0 +1,33 @@
+            _productsManager = productsManager;
+        }
+
+        public IEnumerable<int> ValidateIds(int[] ids)
```
The name "ValidateIds" is not descriptive enough. The actual purpose of this class, according to implementation is to "Validate product ids for order creation". Also this class do nothing if receive invalid Ids. 
I think it`s better for this class to:
1) return nothing. Because commonly Validator classes return nothing or return object, which describes validation errors
2) throw exception if unsupported id was received

k1llama committed:
Fix review comments:

Interfaces renamed to Common;
Change route in controllers;
Removed the ability to add products;
Change return type in MakeOrder method;
Method GetAllGoods renamed to GetAllProducts;
Change input parameter and return type of ValidateIds method;

anton-beetroot-23071996 commented on App.Goods/Services/ValidateService.cs:
```diff
@@ -0,0 +1,31 @@
+            _productsManager = productsManager;
+        }
+
+        public void ValidateIds(List<int> ids)
```
Method name and what it actually did is not correspondent. In most cases "Validators" checks model or some input for validity and then return the validity state somehow, either return boolean or throwing exception.
In your case method is not doing validation, it is doing cleaning. So the valid name should be "Cleaner".
Or do as I wrote before - throw exception when receiving unexpected productId

k1llama committed:
Fix review comments (2):

Rename method ValidateIds to CleanIds

k1llama commented on App.Goods/Services/IValidateService.cs:
```diff
@@ -0,0 +1,9 @@
+
+namespace App.Goods.Services
+{
+    public interface IValidateService
```
Reanme validate service to normalize service. Because it does perform validation actually

k1llama commented on App.Goods/Services/OrdersManager.cs:
```diff
@@ -0,0 +1,45 @@
+            
+            if (!orderedProducts.Any())
+            {
+                return null;
```
it is better to throw exception here

k1llama committed:
Fix review comments (3):

Rename ValidateService class to NormalizeService

k1llama committed:
Merge branch 'master' into Goods

k1llama committed:
fix tabs in ApiTemplate.sln
